mongodbatlas/fw_data_source_mongodbatlas_project_ip_access_list.go

```diff
@@ -0,0 +1,163 @@
+package mongodbatlas
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	cstmvalidator "github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas/framework/validator"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	projectIPAccessList = "project_ip_access_list"
+)
+
+type ProjectIPAccessListDS struct {
+	client *MongoDBClient
+}
+
+func NewProjectIPAccessListDS() datasource.DataSource {
+	return &ProjectIPAccessListDS{}
+}
+
+var _ datasource.DataSource = &ProjectIPAccessListDS{}
+var _ datasource.DataSourceWithConfigure = &ProjectIPAccessListDS{}
+
+func (d *ProjectIPAccessListDS) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = fmt.Sprintf("%s_%s", req.ProviderTypeName, projectIPAccessList)
+}
+
+func (d *ProjectIPAccessListDS) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	client, err := ConfigureClient(req.ProviderData)
+	if err != nil {
+		resp.Diagnostics.AddError(errorConfigureSummary, err.Error())
+		return
+	}
+
+	d.client = client
+}
+
+type tfProjectIPAccessListDSModel struct {
+	ID               types.String `tfsdk:"id"`
+	ProjectID        types.String `tfsdk:"project_id"`
+	CIDRBlock        types.String `tfsdk:"cidr_block"`
+	IPAddress        types.String `tfsdk:"ip_address"`
+	AWSSecurityGroup types.String `tfsdk:"aws_security_group"`
+	Comment          types.String `tfsdk:"comment"`
+}
+
+func (d *ProjectIPAccessListDS) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"project_id": schema.StringAttribute{
+				Required: true,
+			},
+			"cidr_block": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Validators: []validator.String{
+					cstmvalidator.ValidCIDR(),
+					stringvalidator.ConflictsWith(path.Expressions{
+						path.MatchRelative().AtParent().AtName("aws_security_group"),
+						path.MatchRelative().AtParent().AtName("ip_address"),
+					}...),
+				},
+			},
+			"ip_address": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Validators: []validator.String{
```

Comment:
This should have validation for IP address - https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/data_source_mongodbatlas_project_ip_access_list.go#L54

Reply:
I will add a new validator

Reply:
Updated